trie-db/src/triedb.rs

```diff
@@ -95,7 +95,7 @@ where
 	/// may require a database lookup. If `is_root_data` then this is root-data and
 	/// is known to be literal.
 	fn get_raw_or_lookup(&'db self, node: &[u8], partial_key: &[u8]) -> Result<Cow<'db, DBValue>, H::Out, C::Error> {
-		match (partial_key.is_empty(), C::try_decode_hash(node)) {
+		match (partial_key == nibbleslice::EMPTY_ENCODED, C::try_decode_hash(node)) {
```

Comment:
a bit beyond the scope of the PR, but is this `match` assuming that only empty partial-keys could lead to nodes small enough to fit in the hash? A small partial key would easily break that if so

Reply:
Previously (before #12) there was a check for root node here.

Reply:
i see, maybe a comment about that?